### PR TITLE
Error pages for pages being served by Play

### PR DESCRIPTION
**Pull request by andrewconner**
@stkem @eishay Please review. This changes behavior slightly.

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/FortyTwoGlobal.scala [hunk 1]
```diff
@@ -130,14 +131,27 @@ abstract class FortyTwoGlobal(val mode: Mode.Mode)
...
     log.warn(msg)
+
     if (mode == Mode.Test) {
       throw new Exception(s"error [$msg] on $request")
     }
-    Future.successful(allowCrossOrigin(request, BadRequest(msg)))
+
+    if (request.path.startsWith("/internal/")) {
+      allowCrossOrigin(request, BadRequest(msg))
+    } else if (request.path.startsWith("/site/") || request.path.startsWith("/m/") || request.path.startsWith("/ext/") || request.path.startsWith("/search/")) {
```
Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/FortyTwoGlobal.scala]: shouldn't we just check if the requests accepts json? This list is going to be hard to stay on top off (missing eliza, for instance)

File: kifi-backend/modules/common/app/com/keepit/FortyTwoGlobal.scala [hunk 2]
```diff
@@ -145,10 +159,22 @@ abstract class FortyTwoGlobal(val mode: Mode.Mode)
...
+  override def onHandlerNotFound(request: RequestHeader): Future[Result] = Future.successful {
     val errorId = ExternalId[Exception]()
-    log.warn("Handler Not Found %s: on %s".format(errorId, request.path))
-    Future.successful(allowCrossOrigin(request, NotFound("NO HANDLER: %s".format(errorId))))
+    val msg = s"Handler Not Found ${errorId.id}: on ${request.method} ${request.path}"
+    log.warn(msg)
+
+    if (request.path.startsWith("/internal/")) {
+      allowCrossOrigin(request, NotFound(msg))
+    } else if (request.path.startsWith("/site/") || request.path.startsWith("/m/") || request.path.startsWith("/ext/") || request.path.startsWith("/search/")) {
```
Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/FortyTwoGlobal.scala]: same as above

**Comment by stkem** [comment]
made a comment on detecting api requests (as opposed to page requests). lgtm otherwise

File: kifi-backend/modules/common/app/com/keepit/FortyTwoGlobal.scala [hunk 3]
```diff
@@ -130,14 +131,27 @@ abstract class FortyTwoGlobal(val mode: Mode.Mode)
...
+    if (request.path.startsWith("/internal/")) {
+      allowCrossOrigin(request, BadRequest(msg))
+    } else if (speaksJson(request)) {
+      allowCrossOrigin(request, BadRequest(Json.obj(
+        "error" -> errorId.id,
+        "status" -> "bad_request",
+        "request" -> s"${request.method}:${request.path} query: ${request.queryString.mkString("::")}"
+      )))
+    } else {
+      allowCrossOrigin(request, BadRequest(views.html.error.internalError(errorId.id)))
```
Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/FortyTwoGlobal.scala:153]: nice

**Comment by eishay** [comment]
lgtm

**Comment by andrewconner** [comment]
Letting Ray get his actions stuff in first, then will do this one.

**Pull request merged**